### PR TITLE
Create a Subtract bloq

### PR DESCRIPTION
**Pull request by NoureldinYosri**
related to https://github.com/quantumlib/Qualtran/issues/235 since to do comparison in N Toffolis we need to be able to do subtraction

### CODE REVIEW

**Comment by mpharrigan** [comment]
I don't think the relative -> absolute is the correct fix for this particular issue

**Comment by mpharrigan** [comment]
Can you take a look at #933, revert the imports change, and resolve conflicts?

File: qualtran/bloqs/arithmetic/addition.py [hunk 1]
```diff
@@ -409,7 +409,7 @@ class SimpleAddConstant(Bloq):
         [Improved quantum circuits for elliptic curve discrete logarithms](https://arxiv.org/abs/2001.09580) Fig 2a
     """
 
-    bitsize: int
+    bitsize: Union[int, sympy.Expr]
```
Comment (fdmalone) [inline qualtran/bloqs/arithmetic/addition.py:412]: We have a SymbolicInt in qualtran.symbolics.types now.

File: qualtran/bloqs/arithmetic/addition.py [hunk 2]
```diff
@@ -629,3 +633,145 @@ def _add_k_large() -> AddConstantMod:
...
+@frozen
+class Subtract(Bloq):
+    r"""An n-bit subtraction gate.
+
+    Implements $U|a\rangle|b\rangle \rightarrow |a\rangle|a-b\rangle$ using $4n - 4 T$ gates.
+
+    Args:
+        a_dtype: Quantum datatype used to represent the integer a.
+        b_dtype: Quantum datatype used to represent the integer b. Must be large
+            enough to hold the result in the output register of a + b, or else it simply
```
Comment (fdmalone) [inline qualtran/bloqs/arithmetic/addition.py:647]: nit: a - b

**Review by fdmalone — APPROVED** [review]

**Pull request merged**